pipeline/DESIGN.md

```diff
@@ -0,0 +1,212 @@
+| stream_cid  | bytes             | Cid of the stream                                           |
+| stream_type | u8                | Type of the stream, see [stream type values](#stream-types) |
+| controller  | string            | Controller of the stream                                    |
+| dimensions  | map(string,bytes) | Dimensions of the stream                                    |
+
+## chain_proofs
+
+The chain_proofs table contains a row for each on chain proof.
+
+TBD how this table is populated and its schema.
+
+## conclusion_events
+
+The conclusion_events table contains a row for each event in a stream and represents a raw input event after various conclusions have been made.
+
+### Features
+
+* Access to events allowing in order access to event within stream
+* Access to pre-aggregated data for users building their own aggregation system
+* Validation of the event signatures
+
+### Schema
+
+| Column      | Type              | Description                                                                                     |
+| ------      | ----              | -----------                                                                                     |
+| index       | u64               | Order of this event. Index is always greater than the index of any previous event in the stream |
+| stream_cid  | bytes             | Cid of the stream                                                                               |
+| stream_type | u8                | Type of the stream, see [stream type values](#stream-types)                                     |
+| controller  | string            | Controller of the stream                                                                        |
+| dimensions  | map(string,bytes) | Set of key values dimension pairs of the stream                                                 |
+| event_cid   | bytes             | Cid of the event                                                                                |
+| event_type  | u8                | Type of the event, see [event type values](#event-types)                                        |
+| data        | bytes             | DAG-JSON encoding of the event payload                                                          |
+| previous    | list(bytes)       | Ordered list of CID previous to this event. Meaning of the order is stream type dependent       |
+
+
+### Transformation
+
+Raw events are transformed into a flattened structure where stream dimensions and conclusions about data are added to the data.
+
+Conclusions include:
+
+* The event has a valid signature
+* The dimensions and controller of the event
+
+This table joins the raw_events, chain_proofs, and streams tables in order to make the conclusions about the raw events.
+
+## model_schemas
+
+The model_schemas table contains a row for each model and contains the complete resolved schema of the model.
```

Comment:
each model... known about by the node?  Presumably we don't plan to keep indexing all Models forever